crates/widgets/src/slider.rs

```diff
@@ -0,0 +1,212 @@
+use crate::prelude::*;
+
+#[derive(Copy, Clone)]
+enum SliderAction {
+    Move { mouse_x: f64 },
+}
+
+/// The `SliderState` is used to manipulate the position of the thumb of the slider widget.
+#[derive(Default, AsAny)]
+pub struct SliderState {
+    action: Option<SliderAction>,
+    value: f64
+}
+
+impl SliderState {
+    fn action(&mut self, action: SliderAction) {
+        self.action = Some(action);
+    }
+
+    fn adjust(&mut self, ctx: &mut Context) {
+        
+    }
+}
+
+impl State for SliderState {
+    fn init(&mut self, _: &mut Registry, ctx: &mut Context) {
+
+    }
+
+    fn update_post_layout(&mut self, _: &mut Registry, ctx: &mut Context) {
+        if let Some(action) = self.action {
+            match action {
+                SliderAction::Move { mouse_x } => {
+                    if *ctx.child("thumb").get::<bool>("pressed") {
+                        let thumb_width = ctx.child("thumb").get::<Rectangle>("bounds").width();
+                        let track_width = ctx.child("track").get::<Rectangle>("bounds").width();
+                        let slider_x = ctx.widget().get::<Point>("position").x;
```

Comment:
What if we were to store these entities in the state, so that there is no need to look up their entity on each layout update?